agent/recorder.go

```diff
@@ -98,45 +106,38 @@ func (r *SpanRecorder) loop() error {
 					}
 				}
 				cTime = time.Now()
-				err, shouldExit := r.SendSpans()
+				err, shouldExit := r.sendSpans()
 				if err != nil {
 					r.logger.Printf("error sending spans: %v\n", err)
 				}
 				if shouldExit {
 					ticker.Stop()
 					r.t.Kill(err)
-					return nil
+					return err
 				}
 			}
 		case <-r.t.Dying():
-			err, _ := r.SendSpans()
+			err, _ := r.sendSpans()
 			if err != nil {
 				r.logger.Printf("error sending spans: %v\n", err)
 			}
 			ticker.Stop()
```

Comment:
We should do a `defer` of this before entering the `for` loop

Reply:
ok